File: nysol_python/mcmd/methods/convTex/db/smp/func_regexpfx.py
# ================================================
# コマンドマニュアル自動作成用サンプルコードデータ
# ================================================
db={}
db['name']='regexpfx'

############################### IDATA
db['idatas']=[]

data={}
data['name']='dat1.csv'
data['text']='''
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
'''
db['idatas'].append(data)

############################### SCRIPTS
db['scripts']=[]

script={}
script['title']='基本例'
script['comment']='''
正規表現 ``c.*a`` に最も長くマッチする部分文字列のプレフィックスを得る。
例えば ``id=4`` では、$str$項目の ``cabbca`` にマッチしており、そのプレフィックス
すなわち ``ba`` を返している。
``regexstr,regexpfx`` と同じ入力データを使っているので、比較すると分かりやすい。
'''
script['sh_code']='''
mcal c='regexpfx($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
'''
script['py_code']='''
nm.mcal(c='regexpfx($s{str},"c.*a")', a='rsl', i="dat1.csv", o="rsl1.csv").run()
'''
script['odatas']='rsl1.csv'
db['scripts'].append(script)

File: olddoc/mcmd/en/examples/func_regexpfx.rb
#!/usr/bin/env ruby
# coding: utf-8
require "./mkTex.rb"

File.open("dat1.csv","w"){|fpw| fpw.write(
<<'EOF'
id,str
1,xcbbbayy
2,xxcbaay
3,
4,bacabbca
EOF
)}

############## Example 1
title="Basic Example"
comment=<<'EOF'
Find out the prefix of the longest substring that matches with the regular expression \verb|c.*a|. For example, in record \verb|id=4|, the function matches \verb|cabbca| in the  column and returns the prefix \verb|ba|. Since the same input data and substring are used in the \verb|regexstr,regexpfx| function, it is easy to compare the results.
EOF
scp=<<'EOF'
more dat1.csv
mcal c='regexpfx($s{str},"c.*a")' a=rsl i=dat1.csv o=rsl1.csv
more rsl1.csv
EOF
run(scp,title,comment)

